src/pudl/transform/ferc714.py

```diff
@@ -533,3 +542,110 @@ def out_ferc714__hourly_planning_area_demand(
         df[columns], table_name="out_ferc714__hourly_planning_area_demand"
     )
     return df
+
+
+@asset(
+    io_manager_key="pudl_io_manager",
+    compute_kind="pandas",
+)
+def core_ferc714__yearly_planning_area_demand_forecast(
+    raw_ferc714__yearly_planning_area_demand_forecast: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform the yearly planning area forecast data per Planning Area.
+
+    Transformations include:
+
+    - Drop/rename columns.
+    - Remove duplicate rows and average out the metrics.
+
+    Args:
+        raw_ferc714__yearly_planning_area_demand_forecast: Raw table containing,
+            for each year and each planning area, the forecasted summer and winter peak demand,
+            in megawatts, and annual net energy for load, in megawatthours, for the next
+            ten years.
+
+    Returns:
+        Clean(er) version of the yearly forecasted demand by Planning Area.
+    """
+    # Clean up columns
+    df = _pre_process(
+        raw_ferc714__yearly_planning_area_demand_forecast,
+        table_name="core_ferc714__yearly_planning_area_demand_forecast",
+    )
+
+    # For any rows with non-unique respondent_id_ferc714/report_year/forecast_year,
+    # group and take the mean measures
+    # For the 2006-2020 data, there were only 20 such rows. In most cases, demand metrics were identical.
+    # But for some, demand metrics were different - thus the need to take the average.
+    logger.info(
+        "Removing non-unique report rows and taking the average of non-equal metrics."
+    )
+    df = (
+        df.groupby(["respondent_id_ferc714", "report_year", "forecast_year"])[
+            ["summer_peak_demand_mw", "winter_peak_demand_mw", "net_demand_mwh"]
+        ]
+        .mean()
+        .reset_index()
+    )
+
+    # Check all data types and columns to ensure consistency with defined schema
+    df = _post_process(
+        df, table_name="core_ferc714__yearly_planning_area_demand_forecast"
+    )
+    return df
+
+
+@dataclass
+class Ferc714CheckSpec:
+    """Define some simple checks that can run on FERC 714 assets."""
+
+    name: str
+    asset: str
+    num_rows_by_report_year: dict[int, int]
+
+
+check_specs = [
+    Ferc714CheckSpec(
+        name="yearly_planning_area_demand_forecast_check_spec",
+        asset="core_ferc714__yearly_planning_area_demand_forecast",
+        num_rows_by_report_year={
+            2006: 1829,
+            2007: 1570,
+            2008: 1540,
+            2009: 1269,
+            2010: 1259,
+            2011: 1210,
+            2012: 1210,
+            2013: 1192,
+            2014: 1000,
+            2015: 990,
+            2016: 990,
+            2017: 980,
+            2018: 961,
+            2019: 950,
+            2020: 950,
```

Comment:
Wow, interesting progression. I guess there's been a lot of consolidation of planning areas over the last couple of decades.

Reply:
Yeah I noticed that too. Pretty interesting.